checkov/common/packaging/version.py

```diff
@@ -0,0 +1,141 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+#  Partial copy of https://github.com/pypa/packaging/blob/21.3/packaging/version.py
+
+from __future__ import annotations
+
+import re
+from typing import Iterator, TYPE_CHECKING
+
+from packaging import version as packaging_version
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+LegacyCmpKey: TypeAlias = "tuple[int, tuple[str, ...]]"
+Version = packaging_version.Version
+
+
+def parse(version: str) -> packaging_version.Version | LegacyVersion:
+    """
+    Parse the given version string and return either a :class:`Version` object
+    or a :class:`LegacyVersion` object depending on if the given version is
+    a valid PEP 440 version or a legacy version.
+    """
+    try:
+        return packaging_version.parse(version)
+    except packaging_version.InvalidVersion:
+        return LegacyVersion(version)
+
+
+class LegacyVersion(packaging_version._BaseVersion):
+    def __init__(self, version: str) -> None:
+        self._version = str(version)
+        self._key = _legacy_cmpkey(self._version)  # type:ignore[assignment]
+
+    def __str__(self) -> str:
+        return self._version
+
+    def __repr__(self) -> str:
+        return f"<LegacyVersion('{self}')>"
+
+    @property
+    def public(self) -> str:
+        return self._version
+
+    @property
+    def base_version(self) -> str:
+        return self._version
+
+    @property
+    def epoch(self) -> int:
+        return -1
+
+    @property
+    def release(self) -> None:
+        return None
+
+    @property
+    def pre(self) -> None:
+        return None
+
+    @property
+    def post(self) -> None:
+        return None
+
+    @property
+    def dev(self) -> None:
+        return None
+
+    @property
+    def local(self) -> None:
+        return None
+
+    @property
+    def is_prerelease(self) -> bool:
+        return False
+
+    @property
+    def is_postrelease(self) -> bool:
+        return False
+
+    @property
+    def is_devrelease(self) -> bool:
+        return False
+
+
+_legacy_version_component_re = re.compile(r"(\d+ | [a-z]+ | \.| -)", re.VERBOSE)
```

Comment:
are the spaces part of the regular expression? 
also, i suggest to add flag for case insensitive

Reply:
the spaces don't seem to do anything for this use case, just kept them as they were. but will add ignorecase to it

Reply:
Are you sure? the space are meaningful in regex. did you check this? if yes, i will approve it

Reply:
yeap, I tested it. It didn't change the output with or without the spaces.

Reply:
in general you are right that spaces are meaningful, but not for what it is used here